File: docs-examples/example-java/src/test/java/io/micronaut/mqtt/docs/consumer/acknowledge/type/AcknowledgeSpec.java
package io.micronaut.mqtt.docs.consumer.acknowledge.type;

import io.micronaut.mqtt.AbstractMQTTTest;
import io.micronaut.context.ApplicationContext;
import org.junit.jupiter.api.Test;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;

public class AcknowledgeSpec extends AbstractMQTTTest {

    @Test
    void testAckingWithAcknowledgement() {
        ApplicationContext applicationContext = startContext();

// tag::producer[]
ProductClient productClient = applicationContext.getBean(ProductClient.class);
productClient.send("message body".getBytes());
productClient.send("message body".getBytes());
productClient.send("message body".getBytes());
productClient.send("message body".getBytes());
// end::producer[]

        ProductListener productListener = applicationContext.getBean(ProductListener.class);

        try {
            await().atMost(5, SECONDS).until(() ->
                    productListener.messageCount.get() == 4
            );
        } finally {
            applicationContext.close();
        }
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/mqtt/docs/consumer/acknowledge/type/AcknowledgeSpec.kt
package io.micronaut.mqtt.docs.consumer.acknowledge.type

import io.micronaut.mqtt.AbstractMQTTTest
import org.awaitility.Awaitility.await
import org.junit.jupiter.api.Test
import java.util.concurrent.TimeUnit

class AcknowledgeSpec : AbstractMQTTTest() {

    @Test
    fun testAcknowledgeArgument() {
        val applicationContext = startContext()

        // tag::producer[]
        val productClient = applicationContext.getBean(ProductClient::class.java)
        productClient.send("body".toByteArray())
        productClient.send("body".toByteArray())
        productClient.send("body".toByteArray())
        productClient.send("body".toByteArray())
        // end::producer[]

        val productListener = applicationContext.getBean(ProductListener::class.java)

        try {
            await().atMost(5, TimeUnit.SECONDS).until {
                productListener.messageCount.get() == 4
            }
        } finally {
            applicationContext.close()
        }
    }
}
